root/static/scripts/edit/utility/dates.js

```diff
@@ -35,37 +46,40 @@ export const isDateValid = function (y, m, d) {
   }
 
   // Invalid number of days based on the year.
-  if (d < 1 || d > 31 || d > getDaysInMonth(y, m)) {
+  if (d < 1 || d > 31 || y != null && m != null && d > getDaysInMonth(y, m)) {
```

Comment:
We have some tests for `isDateValid` in root/static/scripts/tests/utility.js -- can we add a couple that break without this change?

Some `()` are always nice to clarify the grouping even if it doesn't affect the operator precedence. (Apparently this line is exactly 79 chars, hence the added newlines.)
```suggestion
  if (
    d < 1 || d > 31 ||
    (y != null && m != null && d > getDaysInMonth(y, m))
  ) {
```